internal/transport/http_util.go

```diff
@@ -120,8 +118,32 @@ type decodeState struct {
 	statsTags      []byte
 	statsTrace     []byte
 	contentSubtype string
+}
+
+func newDecodeState(serverSide bool, ignoreContentType bool) *decodeState {
```

Comment:
I think it is actual preferable to not have a constructor here.  Consider:

`state := newDecodeState(true, false)`
and
`state := &decodeState{serverSide: true, ignoreContentType: false}`

The latter is more readable.  Unless there's state that needs to always be initialized for correctness, a constructor is not necessary.

Reply:
done